app/services/action_builder.rb

```diff
@@ -20,6 +20,15 @@ def build_action(extra_attrs = {})
   def previous_action
```

Comment:
Pluck's a good idea. Could also do this in a single query (as per @rodrei suggestion)?

With pluck:
```ruby
## Prettier way to do this?
page_ids = if page.campaign
  page.campaign.pages.pluck(:id)
else
  page.id
end

@previous_action ||= Action.find_by(member: member, page_id: page_ids)
```
with join:
```ruby 
@previous_action ||= Action.joins(:page).where(pages: {campaign_id: page.campaign_id}).
  where(member: member, page_id: page.id).first
```